src/tcpedit/edit_packet.c

```diff
@@ -63,7 +63,9 @@ fix_ipv4_checksums(tcpedit_t *tcpedit, struct pcap_pkthdr *pkthdr, ipv4_hdr_t *i
     
 
     /* calc the L4 checksum if we have the whole packet && not a frag or first frag */
-    if (pkthdr->caplen == pkthdr->len && (htons(ip_hdr->ip_off) & IP_OFFMASK) == 0) {
+    if (pkthdr->caplen == pkthdr->len
+            && pkthdr->len > ntohs(ip_hdr->ip_len)
+            && (htons(ip_hdr->ip_off) & IP_OFFMASK) == 0) {
```

Comment:
Not sure I understand the logic of `&& pkthdr->len > ntohs(ip_hdr->ip_len)`. Can there be conditions where the packet length is altered but ip_hdr->ip_len is not? If so, wouldn't it be better to not set `tcpedit->fixcsum` / `needtorecalc` so that this function is not called?

Also, what about IPv6 in line 95?

If this condition remains, we should at least do `(ntohs(ip_hdr->ip_len) + (ip_hdr->ip_hl << 2))` on the right side.

Note that there will be a merge conflict here due to #190. I'll fix that downstream.